config/cht/tasks.js

```diff
@@ -0,0 +1,528 @@
+const extras = require('./nools-extras');
+
+const {
+  today,
+  getNewestPregnancyTimestamp, 
+  getNewestDeliveryTimestamp,
+  isAlive,
+  isFormArraySubmittedInWindow,
+  getDateMS,
+  getDateISOLocal,
+  getTimeForMidnight,
+  getMostRecentLMPDateForPregnancy,
+  addDays,
+  getRecentANCVisitWithEvent,
+  isPregnancyTaskMuted,
+} = extras;
+
+module.exports = [
+
+  //ANC
+  //ANC Home Visit: 12, 20, 26, 30, 34, 36, 38, 40 weeks (Known LMP)
+  {
+    icon: "icon-pregnancy",
+    title: "task.anc.pregnancy_home_visit.title",
+    appliesTo: "reports",
+    appliesToType: ["pregnancy"],
+    appliesIf: function (c, r) {// If LMP date is known
+      const recentLMP = getMostRecentLMPDateForPregnancy(c, r);
+      let x = recentLMP !== null;
+      //console.log('l', new Date(recentLMP));
+      //console.log('applies', x, c.contact.name);
+      return x;
+    },
+
+    resolvedIf: function (c, r, event, dueDate) {
+      //console.log('ddr', dueDate);
+      if (r.reported_date < getNewestDeliveryTimestamp(c)) return true;//delivery form submitted
+      if (r.reported_date < getNewestPregnancyTimestamp(c)) return true;//old pregnancy report
+
+      //miscarriage or abortion
+      if (getRecentANCVisitWithEvent(c, r, 'abortion') || getRecentANCVisitWithEvent(c, r, 'miscarriage')) return true;
+
+      //Due date older than reported day
+      if(dueDate <= getTimeForMidnight(r.reported_date)) return true;
+
+      //clear tasks
+      if (isPregnancyTaskMuted(c)) return true;
+
+      let x = isFormArraySubmittedInWindow(c.reports, ['pregnancy_home_visit'],
```

Comment:
Can improve the readability of this code by splitting the more complex parameters into variables.

Something like 

```
const startTime = Math.max(addDays(dueDate, -event.start).getTime(), r.reported_date);
const endTime = addDays(dueDate, event.end + 1).getTime();
return isFormArraySubmittedInWindow(c.reports, ['pregnancy_home_visit'], startTime, endTime);
```